include/raft.h

```diff
@@ -336,6 +312,22 @@ enum {
 
 /**
  * A single RPC message that can be sent or received over the network.
+ *
+ * The RPC message types all have a `version` and a `reserved` field.
+ * In the libuv io implementation, `version` is filled out during decoding
+ * and is based on the size of the message on the wire, see e.g.
+ * `sizeofRequestVoteV1`. The version number in the RAFT_MESSAGE_XXX_VERSION
+ * macro needs to be bumped every time the message is changed.
+ *
+ * The `reserved` fields are there to leave enough room for future extensions
+ * because messages could be allocated by a client.
```

Comment:
As far as I understand, the `reserved` field is not needed anymore now. The `version` field acts the same as the one that we use for "external" structures (like `raft_io` and `raft_fsm`).

Anything "producing" (i.e. allocating) a message structure must also set the version field. That way the consumer will know what fields are present and what fields are not present and it won't try to access fields beyond what was allocated.

Furthermore, I probably mentioned that these message structures are effectively part for the `raft_io` interface: they are meant to be used only for communication between `raft` core and `raft_io`. So the only place where they can be allocated is either `raft` core or a `raft_io` implementation. They won't be allocated by "normal" final users of the library.